source/Module/STIG/Functions.Checklist.ps1

```diff
@@ -166,219 +255,227 @@ function New-StigCheckList
-        'releaseinfo'    = $xccdfBenchmarkContent.'plain-text'.InnerText
-        'title'          = $xccdfBenchmarkContent.title
-        'uuid'           = (New-Guid).Guid
-        'notice'         = $xccdfBenchmarkContent.notice.InnerText
-        'source'         = $xccdfBenchmarkContent.reference.source
-    }
 
-    foreach ($StigInfoElement in $stigInfoElements.GetEnumerator())
+    #region STIG_iteration
+    foreach ($XccdfPathItem in $XccdfPath)
     {
-        $writer.WriteStartElement("SI_DATA")
 
-        $writer.WriteStartElement('SID_NAME')
-        $writer.WriteString($StigInfoElement.name)
-        $writer.WriteEndElement(<#SID_NAME#>)
+        $writer.WriteStartElement("iSTIG")
 
-        $writer.WriteStartElement('SID_DATA')
-        $writer.WriteString($StigInfoElement.value)
-        $writer.WriteEndElement(<#SID_DATA#>)
+        #region iSTIG/STIG_INFO
 
-        $writer.WriteEndElement(<#SI_DATA#>)
-    }
+        $writer.WriteStartElement("STIG_INFO")
 
-    $writer.WriteEndElement(<#STIG_INFO#>)
+        $xccdfBenchmarkContent = Get-StigXccdfBenchmarkContent -Path $XccdfPathItem
 
-    #endregion STIGS/iSTIG/STIG_INFO
+        $stigInfoElements = [ordered] @{
+            'version'        = $xccdfBenchmarkContent.version
+            'classification' = 'UNCLASSIFIED'
+            'customname'     = ''
+            'stigid'         = $xccdfBenchmarkContent.id
+            'description'    = $xccdfBenchmarkContent.description
+            'filename'       = Split-Path -Path $XccdfPathItem -Leaf
+            'releaseinfo'    = $xccdfBenchmarkContent.'plain-text'.InnerText
+            'title'          = $xccdfBenchmarkContent.title
+            'uuid'           = (New-Guid).Guid
+            'notice'         = $xccdfBenchmarkContent.notice.InnerText
+            'source'         = $xccdfBenchmarkContent.reference.source
+        }
 
-    #region STIGS/iSTIG/VULN[]
+        foreach ($stigInfoElement in $stigInfoElements.GetEnumerator())
+        {
+            $writer.WriteStartElement("SI_DATA")
+            $writer.WriteStartElement('SID_NAME')
+            $writer.WriteString($stigInfoElement.name)
+            $writer.WriteEndElement(<#SID_NAME#>)
+            $writer.WriteStartElement('SID_DATA')
+            $writer.WriteString($stigInfoElement.value)
+            $writer.WriteEndElement(<#SID_DATA#>)
+            $writer.WriteEndElement(<#SI_DATA#>)
+        }
 
-    # Pull in the processed XML file to check for duplicate rules for each vulnerability
-    [xml]$xccdfBenchmark = Get-Content -Path $xccdfPath -Encoding UTF8
-    $fileList = Get-PowerStigFileList -StigDetails $xccdfBenchmark
-    $processedFileName = $fileList.Settings.FullName
-    [xml]$processed = Get-Content -Path $processedFileName
+        $writer.WriteEndElement(<#STIG_INFO#>)
 
-    $vulnerabilities = Get-VulnerabilityList -XccdfBenchmark $xccdfBenchmarkContent
+        #endregion STIGS/iSTIG/STIG_INFO
```

Comment:
Are these region comments needed?

Reply:
It certainly is helpful. The checklist has sections in it that are very specific. Adam put them in when he wrote it and I like them being there so that I can get oriented as to where I'm generating output.